src/node_api.cc

```diff
@@ -933,12 +933,27 @@ napi_status napi_get_last_error_info(napi_env env,
 }
 
 NAPI_NO_RETURN void napi_fatal_error(const char* location,
-                                     const char* message) {
-  node::FatalError(location, message);
+                                     size_t location_len,
+                                     const char* message,
+                                     size_t message_len) {
+  char* location_string = const_cast<char*>(location);
+  char* message_string = const_cast<char*>(message);
+  if (location_len != -1) {
+    location_string = (char*) malloc(location_len * sizeof(char) + 1);
```

Comment:
I thought we were going to use stack allocation, won't the malloc's result in a memory leak ?

Reply:
@sampsongao pointed out that the process will terminate when we call napi_fatal_error to it should be ok.